src/App.js

```diff
@@ -1,75 +1,158 @@
 import React from 'react';
+import cn from 'classnames';
+import TodoList from './TodoList';
+import TodoFooter from './TodoFooter';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+export const FILTERS = {
+  all: 'All',
+  completed: 'Completed',
+  active: 'Active',
+};
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+export class App extends React.Component {
+  state = {
+    inputValue: '',
+    currentFilter: FILTERS.all,
+    todos: [],
+  };
+
+  addTodo = (e) => {
```

Comment:
a чому addTodo обробляє івенти? З назви метода вона має приймати todo і додавати його в масив todo

Reply:
```suggestion
  addTodo = (todo) => this.setState(prev => ({
    todos: [...prev.todos, todo],
    selectedFilter: FILTER_TYPES.all,
  }))
```
Ось так вона має виглядати і нічого більше:)

Reply:
а как ты тогда предлагаешь их добавлять, если форма на onSubmit страницу перезагружает?

Reply:
всё, я понял